### ISSUE TITLE
A way to add new head and only train that head

### ISSUE THREAD

**Issue by Alonelymess** [ultralytics/ultralytics#17713]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I would like to add a new head that produce multilabel classification (or even multilabel object detection). I want to use the pretrained weight from yolo backbone and neck, then just add the new head and train only the head. Is that possible? Also is there a way to train the model with the custom DataLoader from torch because currently the library doesn't support multilabel

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Alonelymess, thank you for reaching out to us with your query about customizing YOLO with a new head for multilabel classification. This is an automated response, but don't worry, an Ultralytics engineer will assist you soon 🚀!

For initial guidance, I recommend checking our [Docs](https://docs.ultralytics.com), where you can find helpful resources and information on using [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) for custom implementations.

To address your inquiry about modifying the model, it's indeed possible to implement custom heads. If your question pertains to a 🐛 bug or unexpected behavior while doing so, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This will enable us to diagnose the problem effectively. For custom training ❓ questions, please include details such as dataset samples and training logs, and refer to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

You can engage with the Ultralytics community in various ways: join our vibrant [Discord](https://ultralytics.com/discord) for real-time discussions 🎧, participate in profound discussions on [Discourse](https://community.ultralytics.com), or connect with others through threads on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Make sure your `ultralytics` package is up to date with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment, along with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/), by executing:

```bash
pip install -U ultralytics
```

## Environments

YOLO is supported in several verified environments with all necessary dependencies like [CUDA](https://developer.nvidia.com/cuda), [CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** providing free GPU access: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- On the **Google Cloud** via Deep Learning VMs. Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Using **Amazon** Deep Learning AMI. Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- With **Docker**. Check the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Should this badge be green, it confirms all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. Our CI tests ensure the correctness of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across various platforms daily.

**Comment by Y-T-G** [comment]
It's possible, but not straightforward. You can refer to [this](https://y-t-g.github.io/tutorials/yolov8n-add-classes/).

**Comment by Alonelymess** [comment]
> It's possible, but not straightforward. You can refer to [this](https://y-t-g.github.io/tutorials/yolov8n-add-classes/).

Thank you for your reference

**Comment by glenn-jocher** [comment]
You're welcome! If you encounter any issues while implementing the new head or have further questions, feel free to reach out in the thread. The YOLO community and documentation are great resources as well!
